backend/dataall/modules/dataset_sharing/services/share_processors/s3_access_point_process_share.py

```diff
@@ -204,18 +215,68 @@ def clean_up_share(
             source_env_group,
             env_group,
         )
-        clean_up = clean_up_folder.delete_access_point(
-            share=share,
-            dataset=dataset
-        )
+        clean_up = clean_up_folder.delete_access_point(share=share, dataset=dataset)
 
         if clean_up:
             clean_up_folder.delete_target_role_access_policy(
-                share=share,
-                dataset=dataset,
-                target_environment=target_environment
+                share=share, dataset=dataset, target_environment=target_environment
             )
             if not dataset.imported or dataset.importedKmsKey:
                 clean_up_folder.delete_dataset_bucket_key_policy(dataset=dataset)
 
         return True
+
+    @classmethod
+    def verify_shares(
+        cls,
+        session,
+        dataset: Dataset,
+        share: ShareObject,
+        share_folders: [DatasetStorageLocation],
+        source_environment: Environment,
+        target_environment: Environment,
+        source_env_group: EnvironmentGroup,
+        env_group: EnvironmentGroup,
+    ) -> bool:
+        log.info("##### Verifying folders shares #######")
+        for folder in share_folders:
+            sharing_item = ShareObjectRepository.find_sharable_item(
+                session,
+                share.shareUri,
+                folder.locationUri,
+            )
+
+            sharing_folder = cls(
+                session,
+                dataset,
+                share,
+                folder,
+                source_environment,
+                target_environment,
+                source_env_group,
+                env_group,
+            )
+
+            try:
+                sharing_folder.check_bucket_policy()
+                sharing_folder.check_target_role_access_policy()
+                sharing_folder.check_access_point_and_policy()
+
+                if not dataset.imported or dataset.importedKmsKey:
+                    sharing_folder.check_dataset_bucket_key_policy()
+            except Exception as e:
+                sharing_folder.folder_errors = [str(e)]
+
+            if len(sharing_folder.folder_errors):
+                ShareObjectRepository.update_share_item_health_status(
+                    sharing_folder.session,
+                    sharing_item,
+                    ShareItemHealthStatus.Unhealthy.value,
+                    " | ".join(sharing_folder.folder_errors),
```

Comment:
If there are multiple error messages we might get a very very long string. I did a quick search for Postgres VARCHAR type and it could hold. It is very hidden in the docs, I only found this stackoverflow thread https://dba.stackexchange.com/questions/189876/size-limit-of-character-varying-postgresql. Saying that it can store a max of `10485760` chars, we should check in our RDS table and see what is exactly we are using

Reply:
Checked in our Aurora DB - the `character_maximum_length` of the `healthMessage` column is `NULL` meaning it has up to the maximum storage allowed which is as specified above

I do not worry we will run into the limit but I can also truncate the string to certain length if we feel like it is best - it will just give less detailed information to the user on the multiple errors seen

Reply:
I think we are pretty safe, if we have N in Aurora I think we can leave it as is